tests/V8Server/v8-analyzers-test.cpp

```diff
@@ -1271,6 +1271,124 @@ TEST_F(V8AnalyzersTest, test_create) {
+    v8::Isolate::CreateParams isolateParams;
+    ArrayBufferAllocator arrayBufferAllocator;
+    isolateParams.array_buffer_allocator = &arrayBufferAllocator;
+    auto isolate =
+        std::shared_ptr<v8::Isolate>(v8::Isolate::New(isolateParams),
+                                     [](v8::Isolate* p) -> void { p->Dispose(); });
+    ASSERT_TRUE((nullptr != isolate));
+    v8::Isolate::Scope isolateScope(isolate.get());  // otherwise v8::Isolate::Logger() will fail (called from v8::Exception::Error)
+    v8::internal::Isolate::Current()->InitializeLoggingAndCounters();  // otherwise v8::Isolate::Logger() will fail (called from v8::Exception::Error)
+    v8::HandleScope handleScope(isolate.get());  // required for v8::Context::New(...), v8::ObjectTemplate::New(...) and TRI_AddMethodVocbase(...)
+    auto context = v8::Context::New(isolate.get());
+    v8::Context::Scope contextScope(context);  // required for TRI_AddMethodVocbase(...)
+    std::unique_ptr<TRI_v8_global_t> v8g(TRI_CreateV8Globals(isolate.get(), 0));  // create and set inside 'isolate' for use with 'TRI_GET_GLOBALS()'
+    v8g->ArangoErrorTempl.Reset(isolate.get(), v8::ObjectTemplate::New(isolate.get()));  // otherwise v8:-utils::CreateErrorObject(...) will fail
+    v8g->_vocbase = &vocbase;
+    arangodb::iresearch::TRI_InitV8Analyzers(*v8g, isolate.get());
+
+    auto v8Analyzers = v8::Local<v8::ObjectTemplate>::New(isolate.get(), v8g->IResearchAnalyzersTempl)
+                           ->NewInstance();
+    auto fn_create =
+        v8Analyzers->Get(TRI_V8_ASCII_STRING(isolate.get(), "save"));
+    EXPECT_TRUE((fn_create->IsFunction()));
+
+    std::vector<v8::Local<v8::Value>> args = {
+        TRI_V8_STD_STRING(isolate.get(), vocbase.name() + "::testAnalyzer3"s),
+        TRI_V8_ASCII_STRING(isolate.get(), "identity"),
+        TRI_V8_ASCII_STRING(isolate.get(), "{\"abc\":1}")
+    };
+
+    arangodb::auth::UserMap userMap;  // empty map, no user -> no permissions
+    auto& user =
+        userMap
+            .emplace("", arangodb::auth::User::newUser("", "", arangodb::auth::Source::LDAP))
+            .first->second;
+    user.grantDatabase(vocbase.name(), arangodb::auth::Level::RW);  // for system collections User::collectionAuthLevel(...) returns database auth::Level
+    userManager->setAuthInfo(userMap);  // set user map to avoid loading configuration from system database
+
+    auto result = v8::Function::Cast(*fn_create)
+                      ->CallAsFunction(context, fn_create,
+                                       static_cast<int>(args.size()), args.data());
+    EXPECT_TRUE((!result.IsEmpty()));
+    EXPECT_TRUE((result.ToLocalChecked()->IsObject()));
+    auto* v8Analyzer = TRI_UnwrapClass<arangodb::iresearch::IResearchAnalyzerFeature::AnalyzerPool>(
+        result.ToLocalChecked()->ToObject(TRI_IGETC).FromMaybe(v8::Local<v8::Object>()),
+        WRP_IRESEARCH_ANALYZER_TYPE, TRI_IGETC);
+    EXPECT_TRUE((false == !v8Analyzer));
```

Comment:
I see that this is copied from the existing tests, but it's awful and unreadable. Please change it to something more readable like
```suggestion
    EXPECT_NE(nullptr, v8Analyzer);
```
or
```suggestion
    EXPECT_TRUE(v8Analyzer != nullptr);
```
(and also in the other places)

Reply:
That`s a long story of legacy checks. We are planning a separate work for fixing this stuff i all tests

Reply:
OK